docker/docker-compose.yml

```diff
@@ -2,9 +2,9 @@ services:
   path_gateway:
```

Comment:
Why have two docker compose files? We can always up/down specific services regardless.

Reply:
My reasoning is that, with database enabled, the `path_gateway` container depends on the `db` container being up and healthy, ie. uses the `depends_on` property. Once we enable throughput limiting it will also depend upon at least one more dependency. 

To the best of my knowledge there is no way to selectively enable `depends_on`, meaning there are two less than ideal options:
1. Use `depends_on`, meaning that `path_gateway` will always run `db` et al, even when a user only wants to run `path_gateway` on its own (ie without user data/database).
2. Don't use `depends_on`. This means when we want to run all the deps we need to first start up on `db` container then manually wait an arbitrary time (eg. using `sleep _n_`) and then start up `path_gateway` to ensure PATH can connect to the DB on startup. This is doable but inelegant, requiring either a script or long awkward Make targets instead of the built-in `depends_on` solution provided by docker compose.

Using a separate docker compose allows both PATH on its own and PATh + user data to start up gracefully, ensuring that all PATH deps are running and healthy before the container itself runs.

Reply:
I asked ChatGPT about this (see screenshot below) and it seems there is a `--no-deps` [1] option:

![Screenshot 2024-09-16 at 11 12 13 AM](https://github.com/user-attachments/assets/cd14d34a-cfeb-44d7-99e8-2d102f12a39b)

I like to bias to having less code (lower surface area for both maintenance and bugs, less cognitive overhead) and simply support this by having an extra make target.

[1] https://docs.docker.com/reference/cli/docker/compose/up/


![Screenshot 2024-09-16 at 11 11 35 AM](https://github.com/user-attachments/assets/7a6b5860-7aac-4aed-866a-31cdcd27f922)

Reply:
SGTM (and TIL). Not sure how I never came across the `no-deps` flag as I googled and ChatGPTed about it repeatedly. Now that I know I agree it's better to not have multiple docker compose files.

Reply:
Have updated to use only one `docker-compose.yml` and added separate make targets for PATh with & without auth components.